testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksInsertCompatibility.java

```diff
@@ -368,11 +368,11 @@ public void testGeneratedColumnsCompatibility()
                     .containsOnly(row(1, 2), row(2, 4), row(3, 6));
 
             assertQueryFailure(() -> onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES (1, 2, 4)"))
-                    .hasMessageMatching(".*Table default." + tableName + " requires Delta Lake writer version 4 which is not supported");
+                    .hasMessageMatching(".*Writing to tables with generated columns is not supported");
             assertQueryFailure(() -> onTrino().executeQuery("DELETE FROM delta.default." + tableName + " WHERE a_number = 1"))
-                    .hasMessageMatching(".*Table default." + tableName + " requires Delta Lake writer version 4 which is not supported");
+                    .hasMessageMatching(".*Writing to tables with generated columns is not supported");
             assertQueryFailure(() -> onTrino().executeQuery("UPDATE delta.default." + tableName + " SET a_number_times_two = 5 WHERE a_number = 2"))
-                    .hasMessageMatching(".*Table default." + tableName + " requires Delta Lake writer version 4 which is not supported");
+                    .hasMessageMatching(".*Writing to tables with generated columns is not supported");
```

Comment:
Can we merge `testGeneratedColumnsCompatibility` and `testWritesToTableWithGeneratedColumnFails` into one? Sorry, I was missing the existing test in #14513.